Fancy log index

mvollmer opened the pull request.

mvollmer commented:
I have tested this with https://github.com/mvollmer/cockpit/tree/fancy-log-index

stefwalter commented on sink/sink:
```diff
@@ -210,12 +211,21 @@ class Badger(object):
+    def push(self, data):
+        if 'extras' in data:
+            for url in data['extras']:
+                urllib.urlretrieve(url, os.path.basename(url))
```
If this fails, the logging shouldn't fail right? Should we catch exceptions here?

mvollmer replied:
All `push` calls run inside `try` already.  Is that enough?

stefwalter replied:
Ah yes, it does. That's good enough.

stefwalter commented on sink/README.md:
```diff
@@ -71,3 +73,16 @@ The badge itself is a small image that includes `description` and
+last component of the URL will be the name of the file in the log
+directory.
+
+### Index format
```
Can we call this "log" instead of "index"? It makes it easier to understand.

mvollmer replied:
Yes, done. I also added more words.

stefwalter commented on sink/sink:
```diff
@@ -210,12 +211,21 @@ class Badger(object):
+        self.reporters = [ GitHub(config), Irc(config), Badger(config), Extras(config) ]
+        self.dirlink = config.get("Sink", "Url").replace('@@', identifier)
         self.data = None
-        sys.stderr.write(self.link + "\n")
```
The log link still needs to be written for logs that have no status object.

mvollmer replied:
Fixed.

stefwalter commented:
I've posted a cleaner mechanism for the log/index stuff here: https://github.com/cockpit-project/cockpituous/pull/6

What do you think? If you like #6, could you rebase ontop of that?

The 'extra' stuff still makes sense of course.

mvollmer commented:
> What do you think? If you like #6, could you rebase ontop of that?

Yes, very nice.  I have rebased this, updated https://github.com/cockpit-project/cockpit/pull/3082, and test it all.  Works.

mvollmer commented:
Test run: https://mvo.fedorapeople.org/logs/test-6bf1f7cb-fedora-23-x86_64/log.html
